api/envoy/extensions/common/tap/v3/common.proto

```diff
@@ -30,7 +33,12 @@ message CommonExtensionConfig {
     config.core.v3.ConfigSource config_source = 1 [(validate.rules).message = {required: true}];
 
     // Tap config to request from XDS server.
-    string name = 2 [(validate.rules).string = {min_bytes: 1}];
+    string name = 2 [(udpa.annotations.field_migrate).oneof_promotion = "name_specifier"];
+
+    // Resource locator for TAP. This is mutually exclusive to *name*.
+    // [#not-implemented-hide:]
+    udpa.core.v1.ResourceLocator tap_resource_locator = 3
```

Comment:
Similar question here: shouldn't this new field also replace use of the existing `config_source` field?